pilot/pkg/config/kube/crdclient/cache_handler.go

```diff
@@ -88,13 +88,19 @@ func createCacheHandler(cl *Client, schema collection.Schema, i informers.Generi
 	i.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			incrementEvent(kind, "add")
+			if !cl.beginSync.Load() {
```

Comment:
nit: incrementEvent after the `if`?

Reply:
I understand this is to increase for every resource event, no matter whether it triggers push. So i put it before